[GLOBAL]
pants_version = "2.6.0"
colors = true
process_execution_local_parallelism = 4

plugins = [
  "packaging==20.9",
]

backend_packages.add = [
  "pants.backend.python",
  "pants.backend.python.lint.black",
  "pants.backend.python.lint.flake8",
  "pants.backend.python.typecheck.mypy",
  "pants.backend.python.lint.isort",
  "pants.backend.python.lint.docformatter",
  "setup_kwargs_plugin"
]

build_file_prelude_globs = [
  "pants-plugins/macros.py",
]

pythonpath = ["%(buildroot)s/pants-plugins"]

[source]
root_patterns = [
  '/src/python'
]

[python-setup]
interpreter_constraints = ["==3.8.*"]  # can't use 3.9 yet because of https://github.com/intel-isl/Open3D/issues/3065
requirement_constraints = "3rdparty/constraints.txt"
interpreter_search_paths = ["<PATH>", "/usr/bin/python3.8"]

[black]
version = "black==21.7b0"

[flake8]
version = "flake8==3.9.2"
extra_requirements.add = [
  "flake8-tidy-imports",
  "flake8-annotations-coverage",
  "flake8-pytest-style",
  "flake8-bugbear",
  "flake8-pantsbuild",
  "vulture"
]
args = ["--output-file=reports/report.txt"]


[mypy]
version = "mypy==0.910"
extra_requirements.add = [
  "numpy==1.21.1",
  "types-requests==2.25.0",
  "types-aiofiles==0.1.5",
  "types-orjson==0.1.1"
]

[pytest]
pytest_plugins.add = [
  "pytest-repeat",
  "pytest-randomly",
  "pytest-icdiff",
  "pygments",
  "pytest-asyncio"
]

args = ["--count=3 -x"]
timeout_default = 600

[test]
use_coverage = false
extra_env_vars = ["PYTHONDEVMODE=1"]  # "PYTHONWARNINGS=error" not useable due to DeprecationWarning in networkx-2.2

[python-infer]
imports = true
inits = true

[setup-py-generation]
first_party_dependency_version_scheme = "compatible"

[anonymous-telemetry]
enabled = true
repo_id = "394a3b7c-1c30-4047-b53e-90e5df3d84a9"